centreon/www/include/configuration/configObject/timeperiod/DB-Func.php

```diff
@@ -568,3 +386,245 @@ function createTimePeriodsExceptions(array $params): void
     $statement->bindValue(':timeperiod_id', (int) $params['timeperiod_id'], \PDO::PARAM_INT);
     $statement->execute();
 }
+
+// ----------------- API CALLS --------------------
+
+/**
+ * Create a new timeperiod form formData.
+ *
+ * @param array<mixed> $ret
+ *
+ * @return int|null
+ */
+function insertTimePeriodInAPI(array $ret = []): int|null
+{
+    global $form, $basePath;
+
+    $formData = $ret === [] ? $form->getSubmitValues() : $ret;
+
+    try {
+        $timeperiodId = insertTimeperiodByApi($formData, $basePath);
+
+        return $timeperiodId;
+    } catch (Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
+
+        return null;
+    }
+}
+
+/**
+ * Make the API request to create a new timeeperiod and return the new ID.
+ *
+ * @param array $formData
+ * @param string $basePath
+ *
+ * @throws LogicException
+ * @throws Exception
```

Comment:
There is several exceptions that are not in the php docs, you can use this suggestion to fix the alert
```suggestion
 * @throws Throwable
```